aibolit/patterns/multiple_while/multiple_while.py

```diff
@@ -16,9 +15,17 @@ def value(self, filename: str):
         List of LineNumber of methods which have sequential while statements
         """
 
+        tree = AST(build_ast(filename))
         res = []
-        for _, method_node in build_ast(filename).filter(javalang.tree.MethodDeclaration):
-            if len(list(method_node.filter(javalang.tree.WhileStatement))) > 1:
-                res.append(method_node.position.line)
+        nodes = tree.nodes_by_type(ASTNodeType.METHOD_DECLARATION)
+        for node in nodes:
+            cur_line = tree.get_attr(node, 'source_code_line')
+            while_cyclo = tree.all_children_with_type(node, ASTNodeType.WHILE_STATEMENT)
+            count = 0
+            for cyclo in while_cyclo:
```

Comment:
It is better convert generator to list and check its length.
I am currently consider using generators in AST was mine mistake. We might git rid of them soon.

Reply:
ok